### PR TITLE
refactor: remove `BigIntOp`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to https://github.com/flix/flix/issues/5709

### CODE REVIEW

File: main/src/library/Order.flix [hunk 1]
```diff
@@ -337,21 +337,29 @@ instance Order[Int64] {
...
     pub def compare(x: BigInt, y: BigInt): Comparison =
-        if ($BIGINT_LT$(x, y))
-            Comparison.LessThan
-        else if ($BIGINT_GT$(x, y))
-            Comparison.GreaterThan
-        else
-            Comparison.EqualTo
+        import java.math.BigInteger.compareTo(BigInt): Int32 \ {};
+        match x `compareTo` y {
+            case -1 => Comparison.LessThan
```
Comment (magnus-madsen) [inline main/src/library/Order.flix:359]: Are we sure it returns -1? Could it not return any negative number?

**Comment by magnus-madsen** [comment]
Merged, but do check my question :)

File: main/src/library/Order.flix (hunk 1 above)
Comment (jaschdoc) [reply main/src/library/Order.flix:359]: Yes, I checked the documentation, but we can change it to `... if x < 0 => Comparison.LessThan` if you think that's better (objectively it covers the entire input space if we also add the same check for `> 0`)

Comment (magnus-madsen) [reply main/src/library/Order.flix:359]: That at least "feels" safer.

Comment (jaschdoc) [reply main/src/library/Order.flix:359]: True :)

**Pull request merged**